libass/ass_directwrite.c

```diff
@@ -697,20 +697,27 @@ static void match_fonts(void *priv, ASS_Library *lib,
         return;
 
     hr = IDWriteFont_GetFontFamily(font, &fontFamily);
+    IDWriteFont_Release(font);
     if (FAILED(hr) || !fontFamily)
-        goto cleanup;
+        return;
 
-    add_font(font, fontFamily, provider);
+    UINT32 fontCount = IDWriteFontFamily_GetFontCount(fontFamily);
+    for (UINT32 fontIdx = 0; fontIdx < fontCount; ++fontIdx) {
```

Comment:
Couple of style nits: libass uses `snake_case` instead of `camelCase`, so it's better to follow that. Also I think that such long names for variables used in a couple of neighboring lines is overkill: `n` is enough for `fontCount` and `i` for `fontIdx`. And AFAIK we've decided to prefer postfix increment.